tests/Integration/TestCase.php

```diff
@@ -193,4 +277,250 @@ public function go_to_new_post( string $post_status = 'publish' ): int {
+		}
+
+		foreach ( $true_actions as $action ) {
+			self::assertTrue(
+				has_action( $action ),
+				"Unexpected action status: $action should have been called."
+			);
+		}
+
+		foreach ( $false_actions as $action ) {
+			self::assertFalse(
+				has_action( $action ),
+				"Unexpected action status: $action should have not been called."
+			);
+		}
+	}
+
+	/**
+	 * Asserts that a passed script is not registered.
+	 *
+	 * @param string $handle Script handle to test.
+	 */
+	public function assert_is_script_not_registered( string $handle ): void {
+		$this->assert_script_statuses( $handle, array(), array( 'registered' ) );
+	}
+
+	/**
+	 * Asserts that a passed script is registered.
+	 *
+	 * @param string $handle Script handle to test.
+	 */
+	public function assert_is_script_registered( string $handle ): void {
+		$this->assert_script_statuses( $handle, array( 'registered' ) );
+	}
+
+	/**
+	 * Asserts that a passed script is not enqueued.
+	 *
+	 * @param string $handle Script handle to test.
+	 */
+	public function assert_is_script_not_enqueued( string $handle ): void {
+		$this->assert_script_statuses( $handle, array(), array( 'enqueued' ) );
+	}
+
+	/**
+	 * Asserts that a passed script is enqueued.
+	 *
+	 * @param string $handle Script handle to test.
+	 */
+	public function assert_is_script_enqueued( string $handle ): void {
+		$this->assert_script_statuses( $handle, array( 'enqueued' ) );
+	}
+
+	/**
+	 * Asserts multiple enqueuing statuses for a script.
+	 *
+	 * @param string $handle       Script handle to test.
+	 * @param array  $assert_true  Optional. Statuses that should assert to true. Accepts 'enqueued',
+	 *                             'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
+	 * @param array  $assert_false Optional. Statuses that should assert to false. Accepts 'enqueued',
+	 *                             'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
+	 *
+	 * @throws RiskyTestError If no assertions ($assert_true, $assert_false) get passed to the function.
+	 */
+	private function assert_script_statuses( string $handle, array $assert_true = array(), array $assert_false = array() ): void {
+		if ( 0 === count( $assert_true ) + count( $assert_false ) ) {
+			throw new RiskyTestError( 'Function assert_script_statuses() has been used without any arguments' );
+		}
+
+		foreach ( $assert_true as $status ) {
+			self::assertTrue(
+				wp_script_is( $handle, $status ),
+				"Unexpected script status: $handle status should be '$status'"
+			);
+		}
+
+		foreach ( $assert_false as $status ) {
+			self::assertFalse(
+				wp_script_is( $handle, $status ),
+				"Unexpected script status: $handle status should NOT be '$status'"
+			);
+		}
+	}
+
+	/**
+	 * Asserts that a passed style is not registered.
+	 *
+	 * @param string $handle Style handle to test.
+	 */
+	public function assert_is_style_not_registered( string $handle ): void {
```

Comment:
Same as above script code .... the only difference is it verifies the status of styles instead of script.